### ISSUE TITLE
Performance got downgraded in context.Database.EnsureCreatedAsync() after Microsoft.Data.SqlClient upgrade from version 5.1.1 to 5.1.4(5.1.1-->5.1.4) 

### ISSUE THREAD

**Issue by akhiltyagi98** [dotnet/efcore#33847]
### Describe the bug
I am using Microsoft.EntityFrameworkCore localDB SQLServer to run test cases, I use below commands in each test cases

await context.Database.EnsureDeletedAsync().ConfigureAwait(false);
await context.Database.EnsureCreatedAsync().ConfigureAwait(false);

But after upgrading Microsoft.Data.SqlClient package version from 5.1.1 to 5.1.4(5.1.1-->5.1.4), EnsureCreatedAsync() performance downgraded, can you please help me understand this ?

### To reproduce
Please use this code to check

```c#
using Microsoft.EntityFrameworkCore;
using Moq;
using Xunit;

namespace POCSQLClientPerformence.Tests
{
    public class SQLClientPerformenceDB
    {
        private IDbContextFactory<TestDBContext> factory;

        public SQLClientPerformenceDB()
        {
            CreateDbAsync().GetAwaiter().GetResult();
        }

        async Task CreateDbAsync()
        {
            Func<TestDBContext> CreateContext = () =>
            {
                var optionsBuilder = new DbContextOptionsBuilder<TestDBContext>()
                    .UseSqlServer(@$"Server=(localdb)\MSSQLLocalDb;Database=ABCTestDb;Trusted_Connection=True;MultipleActiveResultSets=true");

                return new TestDBContext(optionsBuilder.Options);
            };

            var factoryMock = new Mock<IDbContextFactory<TestDBContext>>();
            factoryMock.Setup(it => it.CreateDbContextAsync(It.IsAny<CancellationToken>())).ReturnsAsync(CreateContext);

            this.factory = factoryMock.Object;
            using var context = await this.factory.CreateDbContextAsync(CancellationToken.None).ConfigureAwait(false);

            await context.Database.EnsureDeletedAsync().ConfigureAwait(false);
            await context.Database.EnsureCreatedAsync().ConfigureAwait(false);
        }

        [Fact]
        public async Task FirstTest()
        {
            // Arrange
            using var context = await this.factory.CreateDbContextAsync(CancellationToken.None).ConfigureAwait(false);

            // Assert - Validate the results or behavior
            Assert.NotNull(context); // Just a dummy assertion for demonstration
        }

        [Fact]
        public async Task SecoundTest()
        {
            // Arrange
            using var context = await this.factory.CreateDbContextAsync(CancellationToken.None).ConfigureAwait(false);

            // Assert - Validate the results or behavior
            Assert.NotNull(context); // Just a dummy assertion for demonstration
        }

        [Fact]
        public async Task ThirdTest()
        {
            // Arrange
            using var context = await this.factory.CreateDbContextAsync(CancellationToken.None).ConfigureAwait(false);

            // Assert - Validate the results or behavior
            Assert.NotNull(context); // Just a dummy assertion for demonstration
        }
    }
 @}
```
SQLClient: 5.2.0(Latest) time:
<img width="960" alt="image" src="https://github.com/dotnet/SqlClient/assets/171185381/5ef067bc-2f14-4cce-a1d1-4906303398d9">

SQLClient: 5.1.1 or later time:
<img width="960" alt="image" src="https://github.com/dotnet/SqlClient/assets/171185381/4cd4e8d8-e44d-447e-9da0-af9f329457cc">

**Mainly` await context.Database.EnsureCreatedAsync().ConfigureAwait(false); `this line taking time. Even If I use context.Database.EnsureCreated() work very quickly.**

**Comment by maumar** [comment]
@akhiltyagi98 are you using sqlite or sql server? The code snippet you provided looks like SqlServer

**Comment by akhiltyagi98** [comment]
@maumar Please check I have attached code in issue description I am using local DB UseSqlServer for testing.

**Comment by maumar** [comment]
Looks like this is known SqlClient issue with hotfix pending in 5.2.1 - https://github.com/dotnet/SqlClient/issues/2542

**Comment by akhiltyagi98** [comment]
@maumar I have opened this issue with SqlClient  if you see discussion they were saying to log issue in EFCore. Can you  open this issue and validate why` await context.Database.EnsureCreatedAsync().ConfigureAwait(false);` taking so much time.

**Comment by maumar** [comment]
reopening now for tracking purposes. @akhiltyagi98 if the performance problem persists after the SqlClient hotfix has been released, please ping me here or on teams and we will take closer look at this

**Comment by ajcvickers** [comment]
Duplicate of #33399

**Comment by akhiltyagi98** [comment]
Fixed in hotfix release 5.2.1.
